### PR TITLE
Add versioned submit block request

### PR DESCRIPTION
**Pull request by avalonche**

### CODE REVIEW

File: spec/versionedsubmitblockrequest.go [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+		if v.Capella.Message == nil {
+			return nil, errors.New("no data message")
+		}
+		return v.Capella.Message.Value, nil
+	default:
+		return nil, errors.New("unsupported version")
+	}
+}
+
+// MessageHashTreeRoot returns the hash tree root of the message of the bid.
```
Comment (mcdee) [inline spec/versionedsubmitblockrequest.go:238]: Not a fan of this, I'd prefer to see a call for `BidTrace()` that returned a `BidTrace` (or `VersionedBidTrace` if you want to future-proof the call) and then users can call `HashTreeRoot()` on that.

File: spec/versionedsubmitblockrequest.go [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+			return 0, errors.New("no data execution payload")
+		}
+		return v.Capella.ExecutionPayload.Timestamp, nil
+	default:
+		return 0, errors.New("unsupported version")
+	}
+}
+
+// Transactions returns the transactions of the payload
+func (v *VersionedSubmitBlockRequest) Transactions() ([]consensusbellatrix.Transaction, error) {
```
Comment (mcdee) [inline spec/versionedsubmitblockrequest.go:476]: I'm a little concerned about this and `Withdrawals()` returning specific versions of their structs.  If there was a `consensusdeneb.Transaction` then the signature of this function would have to change.  But I'm not sure that the alternative of putting in place a versioned structure (or an interface) would be significantly better...

**Review by mcdee — CHANGES_REQUESTED** [review]
Some linting errors as well.

File: spec/versionedsubmitblockrequest.go (hunk 1 above)
Comment (avalonche) [reply spec/versionedsubmitblockrequest.go:238]: I agree, however was following the pattern of other structs such as `VersionedSignedBlindedBeaconBlock()` where the `Root()` function does the same thing

File: spec/versionedsubmitblockrequest.go (hunk 2 above)
Comment (avalonche) [reply spec/versionedsubmitblockrequest.go:477]: I don't see transaction as raw `[]bytes` likely to change but there may be a case for withdrawals. mainly using the transaction field to log the number of transactions and calculate the withdrawals root. Can have functions that calculate those fields but would rather user call `HashTreeRoot()` on the structs.

Comment (mcdee) [reply spec/versionedsubmitblockrequest.go:477]: I think we'll have to go ahead as things are, and accept that the signature may change in future.

**Review by mcdee — APPROVED** [review]

**Pull request merged**